minecraft_java/vanillacord/egg-vanilla-cord.json

```diff
@@ -0,0 +1,45 @@
+    "image": "quay.io\/pterodactyl\/core:java",
+    "startup": "java -Xms128M -Xmx{{SERVER_MEMORY}}M -jar {{SERVER_JARFILE}}",
+    "config": {
+        "files": "{\r\n    \"server.properties\": {\r\n        \"parser\": \"properties\",\r\n        \"find\": {\r\n            \"server-ip\": \"0.0.0.0\",\r\n            \"enable-query\": \"true\",\r\n            \"server-port\": \"{{server.build.default.port}}\",\r\n            \"query.port\": \"{{server.build.default.port}}\"\r\n        }\r\n    }\r\n}",
+        "startup": "{\r\n    \"done\": \")! For help, type \",\r\n    \"userInteraction\": [\r\n        \"Go to eula.txt for more info.\"\r\n    ]\r\n}",
+        "logs": "{\r\n    \"custom\": false,\r\n    \"location\": \"logs\/latest.log\"\r\n}",
+        "stop": "stop"
+    },
+    "scripts": {
+        "installation": {
+            "script": "#!\/bin\/bash\r\n# Vanilla MC Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\napt-get update\r\napt-get install -y curl jq\r\n\r\ncd \/mnt\/server\r\n\r\nLATEST_VERSION=`curl https:\/\/launchermeta.mojang.com\/mc\/game\/version_manifest.json | jq -r '.latest.release'`\r\n\r\nif [ -z \"$VANILLA_VERSION\" ] || [ \"$VANILLA_VERSION\" == \"latest\" ]; then\r\n  INSTALLING_VERSION=$LATEST_VERSION\r\nelse\r\n  INSTALLING_VERSION=$VANILLA_VERSION\r\nfi\r\nMANIFEST_URL=$(curl https:\/\/launchermeta.mojang.com\/mc\/game\/version_manifest.json | jq .versions | jq -r '.[] | select(.id == \"'$INSTALLING_VERSION'\") | .url')\r\n\r\nDOWNLOAD_URL=`curl $MANIFEST_URL | jq .downloads.server | jq -r '. | .url'`\r\n\r\nMAJOR_VERSION=$(echo $INSTALLING_VERSION | grep -Po '^\\K\\d+(?=\\.\\d+\\.\\d+$)')\r\nMINOR_VERSION=$(echo $INSTALLING_VERSION | grep -Po '^\\d+\\.\\K\\d+(?=\\.\\d+$)')\r\nPATCH_VERSION=$(echo $INSTALLING_VERSION | grep -Po '^\\d+\\.\\d+\\.\\K\\d+(?=$)')\r\n\r\nVANILLACORD_URL=https:\/\/src.me1312.net\/jenkins\/job\/VanillaCord\/job\/1.12\/lastSuccessfulBuild\/artifact\/artifacts\/VanillaCord.jar\r\nif [[ MAJOR_VERSION -eq 1 && MINOR_VERSION -lt 12 ]]; then\r\n  VANILLACORD_URL=https:\/\/src.me1312.net\/jenkins\/job\/VanillaCord\/job\/1.7.10\/lastSuccessfulBuild\/artifact\/artifacts\/VanillaCord.jar\r\nfi\r\n\r\nif [[ (MAJOR_VERSION -eq 1 && MINOR_VERSION -eq 7 && PATCH_VERSION -lt 10) || (MAJOR_VERSION -eq 1 && MINOR_VERSION -lt 7) ]]; then\r\n  echo \"VanillaCord is only supported on Minecraft 1.7.10 or higher! You cannot use it with $INSTALLING_VERSION.\"\r\n  exit 1\r\nfi\r\n\r\ncurl -o vanillacord.jar $VANILLACORD_URL\r\njava -jar vanillacord.jar $INSTALLING_VERSION\r\n\r\nrm -f vanillacord.jar\r\nrm -rf in\r\nmv out\/*.jar $SERVER_JARFILE\r\nrm -rf out",
+            "container": "openjdk:8-jre-slim",
```

Comment:
openjdk:8-alpine includes the full JDK. Only the JRE is necessary for the install, openjdk:8-jre-alpine is a smaller image. To be honest, (almost?) all uses of 8-alpine within this project could (should?) be replaced with 8-jre-alpine.

Alpine stock image wouldn't work because this isn't just downloading files. The script is:

```bash
#!/bin/ash

apk --no-cache --update add curl jq

cd /mnt/server

echo $VANILLA_VERSION

LATEST_VERSION=`curl https://launchermeta.mojang.com/mc/game/version_manifest.json | jq -r '.latest.release'`

if { [ -z "$VANILLA_VERSION" ] || [ "$VANILLA_VERSION" == "latest" ]; } then
  INSTALLING_VERSION=$LATEST_VERSION
else
  INSTALLING_VERSION=$VANILLA_VERSION
fi

MAJOR_VERSION=$(echo $INSTALLING_VERSION | sed -En 's/^([0-9]*)\.[0-9]*\.[0-9]*$/\1/p')
MINOR_VERSION=$(echo $INSTALLING_VERSION | sed -En 's/^[0-9]*\.([0-9]*)\.[0-9]*$/\1/p')
PATCH_VERSION=$(echo $INSTALLING_VERSION | sed -En 's/^[0-9]*\.[0-9]*\.([0-9]*)$/\1/p')

VANILLACORD_URL=https://src.me1312.net/jenkins/job/VanillaCord/job/1.12/lastSuccessfulBuild/artifact/artifacts/VanillaCord.jar
if [ $MAJOR_VERSION -eq 1 ] && [ $MINOR_VERSION -lt 12 ]; then
  VANILLACORD_URL=https://src.me1312.net/jenkins/job/VanillaCord/job/1.7.10/lastSuccessfulBuild/artifact/artifacts/VanillaCord.jar
fi

if { [ $MAJOR_VERSION -eq 1 ] && [ $MINOR_VERSION -eq 7 ] && [ $PATCH_VERSION -lt 10 ]; } || { [ $MAJOR_VERSION -eq 1 ] && [ $MINOR_VERSION -lt 7 ]; } then
  echo "VanillaCord is only supported on Minecraft 1.7.10 or higher! You cannot use it with $INSTALLING_VERSION."
  exit 1
fi

curl -o vanillacord.jar $VANILLACORD_URL
java -jar vanillacord.jar $INSTALLING_VERSION

rm -f vanillacord.jar
rm -rf in
mv out/*.jar $SERVER_JARFILE
rm -rf out
```

Note that after we download `vanillacord.jar` we then immediately run it as part of the install script. `vanillacord.jar` is a installer/patcher for the server. vanillacord.jar downloads the Vanilla Minecraft server and then patches it to add support for IP Forwarding. This generates an output jar within the output folder, which we then rename/move to be the server jarfile and delete the installer.

The really important lines are:

```bash
curl -o vanillacord.jar $VANILLACORD_URL
java -jar vanillacord.jar $INSTALLING_VERSION # This is why we need Java
rm -f vanillacord.jar
mv out/*.jar $SERVER_JARFILE
```